common/src/chain/block/block_v1.rs

```diff
@@ -20,7 +27,42 @@ pub struct BlockV1 {
     pub(super) transactions: Vec<Transaction>,
 }
 
+impl BlockVersion for BlockV1 {
+    const BLOCK_VERSION: u32 = 1;
+}
+
+impl Idable<BlockHeader> for BlockHeader {
+    fn get_id(&self) -> Id<Self> {
+        Id::new(&id::hash_encoded(self))
+    }
+}
+
+impl From<&Id<BlockHeader>> for Id<Block> {
+    fn from(id: &Id<BlockHeader>) -> Self {
+        Id::new(&id.get())
+    }
+}
+
+impl From<Id<BlockHeader>> for Id<Block> {
+    fn from(id: Id<BlockHeader>) -> Self {
+        Id::new(&id.get())
+    }
+}
+
 impl BlockV1 {
+    pub fn check_version(&self) -> Result<(), super::BlockConsistencyError> {
+        let a = self.header.block_version;
+        let b = <Self as BlockVersion>::static_version(self);
+        if a != b {
+            return Err(super::BlockConsistencyError::VersionMismatch(a, b));
+        }
+        Ok(())
+    }
```

Comment:
Hmm wondering if we could just have the version in one place rather than having header version and block version and requiring them to be in agreement.

Reply:
That's exactly the struggle. Given that we want a separate Header + we want this to be handled with enums, it's not really possible. In other words, we cannot use the version byte from the enum in the hash because the header alone doesn't know about the enum. The only solution there is to get rid of the enum and then process the block contents completely manually.